### DISCUSSION TITLE
SSG with an unknown number of dynamic routes

### DISCUSSION THREAD

**Discussion by joshfarrant** [vercel/next.js#24201] [Help]
Hi all,

I'm in the process of building a portal that interacts with an API. I'm running into a few issues around dynamic routes when running the `next export`ed build on Netlify.

The app follows this style of URL structure: `myapp.com/customers/[:customerId]/things/[:thingId]`. When I try to navigate directly to one of those pages I'm hitting a 404, presumably as there's no file corresponding to the specific parameters in the URL (customerID = 123, thingId = 456).

As the API has an unknown number of _customers_ and _things_ I can't generate routes for all of them at build time. Is there any way that I can set up redirects to do something like:

> When the user navigates to `/customers/123/things/456` preserve the URL but actually serve the file `pages/customers/[customerId]/things/[thingId]`?

I remember that this sort of thing was possible in the CRA days (I suppose because the whole app was bundled into a single file), but I'm not sure of the best way to achieve it in Next.

Thanks!

**Comment by imadatyatalah** [comment]
As you've said that you have unknown number of dynamic routes, The best thing you can do is to set `fallback` to `true` in your `getStaticPaths` function.

Because if `fallback` is `false`, then any paths not returned by `getStaticPaths` will result in a 404 page (That's why you've got error 404).

`fallback: true` is not supported when using `next export`.

Basic example:
```jsx
export const getStaticPaths = async () => {
  return { paths: [], fallback: true };
};
```

[Learn more about `fallback: true`](https://nextjs.org/docs/basic-features/data-fetching#fallback-true)
[When is `fallback: true` useful?](https://nextjs.org/docs/basic-features/data-fetching#when-is-fallback-true-useful)
